tests/Yaapii.Atoms.Tests/Map/MapEnvelopeTest.cs

```diff
@@ -41,16 +41,58 @@ public void TryGetValueWithExistingKey()
         [Fact]
         public void TryGetValueWithMissingKey()
         {
-
             var map = new NonAbstractEnvelope(new Dictionary<int, int> { { 7, 42 } });
             int outValue;
             Assert.False(map.TryGetValue(0, out outValue));
         }
 
+        [Fact]
+        public void GetValueWithExistingKey()
+        {
+            var map = new NonAbstractEnvelope(new Dictionary<int, int> { { 7, 42 } });
+            var outValue = map[7];
+            Assert.Equal(42, outValue);
+        }
+
+        [Fact]
+        public void GetValueWithMissingKey()
+        {
+            var map = new NonAbstractEnvelope(new Dictionary<int, int> { { 7, 42 } });
+
+            try
+            {
+                var outValue = map[0];
+                Assert.True(false, "This has to fail");
+            }
+            catch (ArgumentException ex)
+            {
+                Assert.Equal("The key '0' is not present in the map.", ex.Message);
```

Comment:
Why doesnt Xunit has an Assert.Throws which compares the `ex.Message`?

Reply:
Ask them ;) Maybe you could make one by implementing an extension to Assert.

Reply:
```csharp
var ex = Assert.Throws(() => {...})
Assert.Equal("foo bar", ex.Message)
```